homeassistant/components/google/calendar.py

```diff
@@ -147,18 +140,30 @@ async def async_get_events(self, hass, start_date, end_date):
         params["timeMin"] = start_date.isoformat("T")
         params["timeMax"] = end_date.isoformat("T")
 
+        event_list = []
         events = await hass.async_add_executor_job(service.events)
+        page_token = None
+        while True:
+            page_token = await self.async_get_events_page(
```

Comment:
Why do we need to fetch so many events to begin with?

Reply:
Because there are so many. People that use Google Calendar for work easily have 6 appointments on an average working day, resulting on 360 appointments for a 6 week calendar page.
Setting a max won't be helpful, as the sett of omitted appointments is a random one, so that is always undesirable. The original implementation allows increasing ``max_results`` to Google's max: 2500. That however might cause timeouts and it will cause excess memory use.
From a memory perspective, we might even want to a lower page size than Google's default (250) and e.g. fetch pages of 100 appointments.

Reply:
Now uses a page size of 100